inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationScopeResolver.java

```diff
@@ -0,0 +1,257 @@
+     */
+    private ElementMatcher.Junction<MethodDescription> buildMethodMatcher(InstrumentationScopeSettings scopeSettings, ElementMatcher.Junction<TypeDescription> typeMatcher) {
+        MatcherBuilder<MethodDescription> builder = new MatcherBuilder<>();
+
+        if (scopeSettings.getMethodScope() != null) {
+            scopeSettings.getMethodScope().forEach(m -> processMethod(builder, m));
+        }
+
+        if (builder.isEmpty()) {
+            builder.and(any());
+        }
+
+        ElementMatcher.Junction<MethodDescription> matcher = builder.build();
+
+        if (scopeSettings.getAdvanced() != null && scopeSettings.getAdvanced().getInstrumentOnlyInheritedMethods()) {
+            MatcherBuilder<TypeDescription> superBuilder = new MatcherBuilder<>();
+
+            if (scopeSettings.getTypeScope().getInterfaces() != null) {
+                scopeSettings.getTypeScope()
+                        .getInterfaces()
+                        .forEach(i -> superBuilder.or(buildNameMatcher(i)));
+            }
+            if (scopeSettings.getTypeScope().getSuperclass() != null) {
+                superBuilder.or(buildNameMatcher(scopeSettings.getTypeScope().getSuperclass()));
+            }
+
+            if (!superBuilder.isEmpty()) {
+                matcher = isOverriddenFrom(superBuilder.build()).and(builder.build());
+            }
+        }
+
+        return matcher;
+    }
+
+    /**
+     * Processing a single {@link MethodMatcherSettings} and adding it to the given {@link MatcherBuilder}.
+     */
+    private void processMethod(MatcherBuilder<MethodDescription> builder, MethodMatcherSettings matcherSettings) {
+        MatcherBuilder<MethodDescription> innerBuilder = new MatcherBuilder<>();
+
+        innerBuilder.and(matcherSettings.getIsConstructor(), isConstructor());
+        innerBuilder.and(buildVisibilityMatcher(matcherSettings.getVisibility()));
+        innerBuilder.and(buildArgumentMatcher(matcherSettings.getArguments()));
+
+        if (!matcherSettings.getIsConstructor()) {
+            ElementMatcher.Junction<MethodDescription> nameMatcher = buildNameMatcher(matcherSettings);
+            innerBuilder.and(nameMatcher);
+            if (matcherSettings.getIsSynchronized() != null) {
+                innerBuilder.and(matcherSettings.getIsSynchronized(), isSynchronized());
+            }
+        }
+
+        builder.or(innerBuilder.build());
+    }
+
+    /**
+     * Creating an {@link ElementMatcher} matching methods which takes arguments equals to the given array. The array consists
+     * of String representing full qualified class names.
+     */
+    private ElementMatcher.Junction<MethodDescription> buildArgumentMatcher(String[] arguments) {
+        if (arguments == null) {
+            return null;
+        } else if (arguments.length == 0) {
+            return takesArguments(0);
+        } else {
+            MatcherBuilder<MethodDescription> builder = new MatcherBuilder<>();
+

```

Comment:
Here you actually match for a prefix. Imagine you specify two arguments. Some methods takes these two arguments but an additional one, this method is matched too.
Also you initially had an approach with a Map<Integer,String> to specify the argument filters, why did you go for the List in the end? the Map approach gives more flexibility and is not more difficult to write imo

Reply:
Done.